### PR TITLE
Feature/vagrant virtual environment

### PR DESCRIPTION
**Pull request by Rojo**
This allows running the software in a virtual machine with Vagrant and Virtual Box for development and/or testing purposes.

* Updated the Vagrant file and included the necessary provisioning scripts for all the required software.
* Included configuration files and instructions on how to run the software within a virtual machine.
* Changed the `config/environments/development.rb` settings so the software performs well under a virtual machine. (Otherwise it works unbearable slow).
* Updated the Ruby version to the 2.3.6 release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -290,6 +291,77 @@ See the [Open Projects](#open-projects) section for a detailed description of th
...
+**Note:** The provision scripts configure the sofware dependencies to play nice with each other while
+they are inside the same virtual environment but said configuration is not optimized for a production
+environment.
+
+1. Install [Vagrant][vagrant] and [Virtual Box][virtualbox] (with the extension package).
+
+2. Retrieve the project from Git
+
+   ```bash
+   git clone https://github.com/MakerNetwork/MakerNet.work.git
```
Comment (sylvainbx) [inline README.md]: You may point this URL to our repository

File: .ruby-version [hunk 2]
```diff
@@ -1 +1 @@
-ruby-2.3.0
+ruby-2.3.6
```
Comment (sylvainbx) [inline .ruby-version:1]: Why is this needed?

File: README.md [hunk 3]
```diff
@@ -290,6 +291,77 @@ See the [Open Projects](#open-projects) section for a detailed description of th
...
+
+   ```bash
+   cp config/database.yml.virtual config/database.yml
+   cp config/application.yml.default config/application.yml
+   ```
+
+10. Set up the databases. (You should provide the desired admin credentials):
+
+   ```bash
+   ADMIN_EMAIL='youradminemail' ADMIN_PASSWORD='youradminpassword' rake db:setup
```
Comment (sylvainbx) [inline README.md]: As stated at point 8 of [Setup a development environment](https://github.com/LaCasemate/fab-manager/blob/master/README.md#general-guidelines) procedure, `rake db:setup` will not propagate the raw SQL instructions needed for members autocompletion and other features.

File: config/environments/development.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
   # In the development environment your application's code is reloaded on
   # every request. This slows down response time but is perfect for development
   # since you don't have to restart the web server when you make code changes.
-  config.cache_classes = false
+  config.cache_classes = true
```
Comment (sylvainbx) [inline config/environments/development.rb]: This configuration must not be the default value as it will complexify software development. You may have to use a environment variable to configure this only if running with vagrant.

File: config/environments/development.rb [hunk 5]
```diff
@@ -25,7 +25,7 @@
   # Debug mode disables concatenation and preprocessing of assets.
   # This option may cause significant delays in view rendering with a large
   # number of complex assets.
-  config.assets.debug = true
+  config.assets.debug = false
```
Comment (sylvainbx) [inline config/environments/development.rb]: Same remark here

**Review by sylvainbx — CHANGES_REQUESTED** [review]
Thanks for your contribution. This is a great idea but the points noticed below must be corrected before I can merge your changes, especially thoses in development.rb

File: .ruby-version (hunk 2 above)
Comment (Rojo) [reply .ruby-version:1]: Just so the users of the software can execute it with the current security and bug fixes of the 2.3 Ruby version.

File: README.md (hunk 1 above)
Comment (Rojo) [reply README.md]: URL fixed!

File: README.md (hunk 3 above)
Comment (Rojo) [reply README.md]: Yikes! I had not seen that before. It has the complete instructions and a warning legend now.

File: .ruby-version (hunk 2 above)
Comment (sylvainbx) [reply .ruby-version:1]: Ok but I will have to do some tests before releasing

**Review by sylvainbx — APPROVED** [review]

**Comment by sylvainbx** [comment]
Thanks, I'll publish this feature into the next release.

**Comment by Rojo** [comment]
Me pleasure. I hope it will be useful for new users of the manager.
